### PR TITLE
fix: sheet getRowVisible lags after filter

### PR DESCRIPTION
**Pull request by lumixraku**
close https://github.com/dream-num/univer/issues/3369

## Pull Request Checklist

- [ ] Related tickets or issues have been linked in the PR description (or missing issue).
- [ ] [Naming convention](https://github.com/dream-num/univer/blob/dev/docs/NAMING_CONVENTION.md) is followed (**do please** check it especially when you created new plugins, commands and resources).
- [ ] Unit tests have been added for the changes (if applicable).
- [ ] Breaking changes have been documented (or no breaking changes introduced in this PR).

### CODE REVIEW

File: packages/core/src/sheets/worksheet.ts [hunk 1]
```diff
@@ -463,19 +464,32 @@ export class Worksheet {
...
+     * @param row
+     * @returns {boolean} is row hidden by filter
+     */
+    isRowFiltered(row: number): boolean {
+        if (!this._filteredRowCacheMap.has(row)) {
+            this._filteredRowCacheMap.set(row, this._viewModel.getRowFiltered(row));
+        }
+        return !!this._filteredRowCacheMap.get(row);
+    }
+
```
Comment (lumixraku) [inline packages/core/src/sheets/worksheet.ts:477]: Key changes, add cache for isRowFiltered, intercetor slow down performance.

Comment (Gggpound) [reply packages/core/src/sheets/worksheet.ts:477]: 为什么这个cache不做到  packages/sheets-filter/src/controllers/sheets-filter.controller.ts 里面？

Comment (Gggpound) [reply packages/core/src/sheets/worksheet.ts:477]: 是interceptor的自身逻辑导致的性能问题？判断是否是filter行，我看底层就是`set.has`这样一个逻辑,看起来这个并不是一个高耗时的逻辑。
这个cache的设计，看起来就是单纯为了避免走一次 interptor 自身的逻辑？

**Review by wzhudev — APPROVED** [review]

File: packages/core/src/sheets/worksheet.ts (hunk 1 above)
Comment (lumixraku) [reply packages/core/src/sheets/worksheet.ts]: rollback 了

**Comment by zhaolixin7** [comment]
清楚筛选后内渲染不出来，无法滑动表格，路径：1、选中整个表格设置筛选，2、仅筛选任意项，3、拖拽冻结线冻结行、4、清楚筛选条件 
[2024_9_11 15_37_45 video.webm](https://github.com/user-attachments/assets/9ad0e800-6ea9-4ba2-8f8e-0f49c25a3ff1)
[2024_9_11 15_40_03 snapshot.json](https://github.com/user-attachments/files/16960542/2024_9_11.15_40_03.snapshot.json)

**Pull request merged**